src/core/Expander/Expander/Expander.tsx

```diff
@@ -0,0 +1,226 @@
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** Expander consumer for open state */
+  consumer: ExpanderProviderState;
+}
+
+interface BaseExpanderProps extends InternalExpanderProps {
+  id: string;
+}
+
+class BaseExpander extends Component<BaseExpanderProps> {
+  state: ExpanderState = {
+    openState: this.props.defaultOpen || false,
+  };
+
+  constructor(props: BaseExpanderProps) {
+    super(props);
+    if (!!props.consumer) {
+      const defaultOpen =
+        props.open !== undefined ? props.open : props.defaultOpen || false;
+      props.consumer.onExpanderOpenChange(props.id, defaultOpen);
+    }
+  }
+
+  componentDidUpdate(prevProps: ExpanderProps, prevState: ExpanderState) {
+    const { consumer, open } = this.props;
+    const controlled = open !== undefined;
+    if (
+      !!consumer &&
+      prevProps.id !== undefined &&
+      prevProps.id !== this.props.id
+    ) {
+      consumer.onExpanderOpenChange(prevProps.id, undefined);
+    }
+    if (
+      !!consumer &&
+      consumer.toggleAllExpanderState !==
+        prevProps.consumer?.toggleAllExpanderState
+    ) {
+      if (
+        (!controlled &&
+          !!this.state.openState !== consumer.toggleAllExpanderState.toState) ||
+        (controlled && open !== consumer.toggleAllExpanderState.toState)
+      ) {
+        this.handleOpenChange();
+      }
+    }
+    if (
+      (!controlled && prevState.openState !== this.state.openState) ||
+      (controlled && prevProps.open !== open)
+    ) {
+      if (!!consumer && this.props.id !== undefined) {
+        const currentState = controlled ? !!open : this.state.openState;
+        consumer.onExpanderOpenChange(this.props.id, currentState);
+      }
+    }
+  }
+
+  componentWillUnmount() {
+    if (!!this.props.consumer && !!this.props.consumer.onExpanderOpenChange) {
+      this.props.consumer.onExpanderOpenChange(this.props.id, undefined);
+    }
+  }
+
+  handleOpenChange = () => {
+    const { open, onOpenChange } = this.props;
+    const { openState } = this.state;
+    const controlled = open !== undefined;
+    const newOpenState = controlled ? !!open : !openState;
+    if (!controlled) {
+      this.setState({ openState: newOpenState });
+    }
+    if (!!onOpenChange) {
+      onOpenChange(newOpenState);
+    }
+  };
+
+  render() {
+    const {
+      id,
+      open,
+      defaultOpen,
+      onOpenChange,
+      className,
+      children,
+      consumer,
+      ...passProps
+    } = this.props;
+    const openState = open !== undefined ? !!open : this.state.openState;
+    const titleId = `${id}_title`;
+    const contentId = `${id}_content`;
+
+    return (
+      <HtmlDiv
+        id={id}
+        {...passProps}
+        className={classnames(className, baseClassName, {
+          [openClassName]: !!openState,
+        })}
+      >
+        <ExpanderProvider
+          value={{
+            open: openState,
+            contentId,
+            titleId,
```

Comment:
Would just the `id` be enough to be here? 
Then using it in later places to make those `{id}_content` and `{id}_title`.

Reply:
I have no preference for either solution in this case since there are so few props that everything fits nicely into one view with both ways and the consts are only used once. The proposed solution has fewer lines so will change.

Reply:
Changed to use id directly since needed only in one place per id.